### PR TITLE
Js task2

### PR DESCRIPTION
**Pull request by aotarasenko**

### CODE REVIEW

File: js_task2/index.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+              </div>
+
+              <button class="add-note" name="add-item" type="button"></button>
+            </fieldset>
+          </form>
+        </div>
+
+        <div class="table">
+          <div class="table-heading is-flex-row">
+            <input type="checkbox" />
```
Comment (marynakorol) [inline js_task2/index.html]: the checkbox should not be a default, please style it according to task image

File: js_task2/index.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                  Filter
+                </button>
+              </div>
+
+              <button class="add-note" name="add-item" type="button"></button>
+            </fieldset>
+          </form>
+        </div>
+
+        <div class="table">
```
Comment (marynakorol) [inline js_task2/index.html:57]: OPTIONALLY: easier way to make it a table - just to use table tags 
<table>    <thead>          <tr>               <th>..

File: js_task2/src/js/ModalWindow.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  modal
+  eventType
+  id
+
+  constructor(user, btnType, id) {
+    this.userFromInput = user
+    this.eventType = btnType
+    this.id = id
+    document.body.appendChild(this.renderModalWindow())
+    this.modal = document.querySelector('#modal')
```
Comment (marynakorol) [inline js_task2/src/js/ModalWindow.js]: better to get element by id

File: js_task2/src/js/dataFunctions.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { baseUrl, outputArea } from './_variables'
+import axios from 'axios'
+import { renderRow } from './renderFunctions'
+
+export const getUsers = async (baseUrl) => {
```
Comment (marynakorol) [inline js_task2/src/js/dataFunctions.js:5]: getting and changing data should work without page reloading

File: js_task2/src/js/index.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+})
+
+addBtn.addEventListener('click', (Event) => {
+  Event.preventDefault()
+  addUser()
+})
+
+mainCheckbox.addEventListener('click', () => {
+  if (mainCheckbox.checked) {
+    selectWholeTable(true)
```
Comment (marynakorol) [inline js_task2/src/js/index.js]: when the main checkbox is checked there should be a possibility to delete all rows not by clicking some of the trash icons. but please show the delete button near the main row
image

File: js_task2/src/scss/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,8 @@
+$font-secondary: rgb(129, 129, 129);
```
Comment (marynakorol) [inline js_task2/src/scss/_variables.scss]: use $color- at the beginning for this var to understand it is a color inside it

File: js_task2/src/scss/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+$background-primary: #6734eb;
+$background-secondary: #eb3464;
+$white: #fff;
```
Comment (marynakorol) [inline js_task2/src/scss/_variables.scss]: use color-light or smth like this

File: js_task2/src/scss/app-style.scss [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+.page-name {
+  padding: 12px;
+  background-color: $background-primary;
+  color: $white;
+  @include app-font;
+}
+
+.filter-section {
+  border-bottom: 1.5px solid rgba(128, 128, 128, 0.171);
```
Comment (marynakorol) [inline js_task2/src/scss/app-style.scss]: 0.17 would be enough and 0.2 would be better for transparency in rgba()

File: js_task2/src/scss/app-style.scss [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+        @include desktop-size {
+          flex-direction: row;
+        }
+        input {
+          padding: 10px 8px;
+          border: 2px solid rgba(214, 214, 214, 0.719);
+          border-radius: 2px;
+          outline: none;
+          margin-bottom: 12px;
+          width: 100%;
```
Comment (marynakorol) [inline js_task2/src/scss/app-style.scss]: please keep box model order, width, and margin should be above

File: js_task2/src/scss/app-style.scss [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+          @include app-font;
+          @include desktop-size {
+            margin-right: 10px;
+            margin-bottom: 0;
+            width: auto;
```
Comment (marynakorol) [inline js_task2/src/scss/app-style.scss:76]: same as 70, 71 lines

File: js_task2/src/scss/app-style.scss [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+        .filter-btn {
+          border: none;
+          border-radius: 2px;
+          padding: 8px 16px;
+          outline: none;
+          @include app-font;
+          text-transform: uppercase;
+          background-color: $background-primary;
+          color: $white;
```
Comment (marynakorol) [inline js_task2/src/scss/app-style.scss]: same as 70 line

File: js_task2/src/scss/app-style.scss [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+.output-data {
+  .user-row {
+    justify-content: space-between;
+    border-bottom: 1px solid $color-neutral;
+    width: 100%;
+
+    &:hover {
+      background-color: lightgrey;
```
Comment (marynakorol) [inline js_task2/src/scss/app-style.scss]: for the future please avoid using these default colors, they can be different in some browsers 
for now you can leave it as it is

File: js_task2/src/scss/general.scss [hunk 13]
```diff
@@ -0,0 +1,10 @@
+.container {
+  max-width: 1440px;
+  min-width: 320px;
```
Comment (marynakorol) [inline js_task2/src/scss/general.scss]: you do not need this value if you use mobile-first approach

File: js_task2/src/scss/modal.scss [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  height: 100%;
+
+  background-color: rgba(128, 128, 128, 0.555);
+
+  form {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    padding: 24px;
+    min-width: 350px;
```
Comment (marynakorol) [inline js_task2/src/scss/modal.scss]: same as 3line in general.scss

it's interesting that you have a form with min-width 350px inside a container with min-width 320px, so it won't fit it

**Review by marynakorol — CHANGES_REQUESTED** [review]
please try to change your code to get data without reloading

File: js_task2/src/js/eventFunctions.js [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+import { setDataToObject } from './helpers';
+import ModalWindow from './ModalWindow';
+
+export const addRowSelectionEvent = (row) => {
+  row.addEventListener('click', (Event) => {
+    if (Event.target.type === 'button') {
+      return;
+    }
+
+    let checkbox = row.querySelector('input[type="checkbox');
```
Comment (marynakorol) [inline js_task2/src/js/eventFunctions.js]: re-check please this selector, it should be 'input[type="checkbox"]'

File: js_task2/src/js/index.js (hunk 5 above)
Comment (aotarasenko) [reply js_task2/src/js/index.js]: done

File: js_task2/src/js/eventFunctions.js (hunk 15 above)
Comment (aotarasenko) [reply js_task2/src/js/eventFunctions.js]: thx, changed

File: js_task2/src/scss/modal.scss (hunk 14 above)
Comment (aotarasenko) [reply js_task2/src/scss/modal.scss]: deleted this

File: js_task2/src/scss/general.scss (hunk 13 above)
Comment (aotarasenko) [reply js_task2/src/scss/general.scss]: done

File: js_task2/src/scss/app-style.scss (hunk 12 above)
Comment (aotarasenko) [reply js_task2/src/scss/app-style.scss]: thx

File: js_task2/src/js/dataFunctions.js (hunk 4 above)
Comment (aotarasenko) [reply js_task2/src/js/dataFunctions.js:5]: now works right)

File: js_task2/src/js/ModalWindow.js (hunk 3 above)
Comment (aotarasenko) [reply js_task2/src/js/ModalWindow.js]: okay, changed

File: js_task2/src/scss/_variables.scss (hunk 6 above)
Comment (aotarasenko) [reply js_task2/src/scss/_variables.scss]: done

File: js_task2/src/scss/_variables.scss (hunk 7 above)
Comment (aotarasenko) [reply js_task2/src/scss/_variables.scss]: done

File: js_task2/src/scss/app-style.scss (hunk 9 above)
Comment (aotarasenko) [reply js_task2/src/scss/app-style.scss]: done

File: js_task2/src/scss/app-style.scss (hunk 11 above)
Comment (aotarasenko) [reply js_task2/src/scss/app-style.scss]: done

**Comment by aotarasenko** [comment]
still have default checkboxes, i will change their while waited an review

File: js_task2/src/scss/_variables.scss (hunk 7 above)
Comment (marynakorol) [reply js_task2/src/scss/_variables.scss]: It's still $white

File: js_task2/index.html [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+            <button
+              class="delete-item-btn"
+              type="button"
+              name="delete"
+            ></button>
+          </div>
+        </div>
+
+        <div class="output-data"></div>
+        <div class="change-page"></div>
```
Comment (marynakorol) [inline js_task2/index.html:77]: style please an active page number with blue bg

File: js_task2/src/js/ModalWindow.js [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+  modal;
+  eventType;
+  id;
+
+  constructor(user, btnType, id) {
+    this.userFromInput = user;
+    this.eventType = btnType;
+    this.id = id;
+    document.body.appendChild(this.renderModalWindow());
+    this.modal = document.getElementById('#modal');
```
Comment (marynakorol) [inline js_task2/src/js/ModalWindow.js]: remove #

File: js_task2/src/js/helpers.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+    row.firstChild.firstChild.checked = state;
+    if (state) {
+      checkedItems.push(row);
+    } else {
+      checkedItems.pop(row);
+    }
+  });
+}
+
+export const changePage = (pageIndex, btn) => {
```
Comment (marynakorol) [inline js_task2/src/js/helpers.js:36]: pagination work's a little incorrectly. when you are on the first page the left button should be disabled, the same behavior for the last page and right button. And add please selected-page class to the first page in advance, cause when you are at the beginning of the table, it assumes that you are at the first page

File: js_task2/src/js/helpers.js [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+    item.classList.remove('selected-page');
+  });
+
+  btn.classList.add('selected-page');
+  checkFilterFields();
+};
+
+export const checkFilterFields = () => {
+  let dataRows = document.querySelectorAll('.user-row');
+  let filterFields = document.querySelectorAll('.filter-inputs input');
```
Comment (marynakorol) [inline js_task2/src/js/helpers.js:88]: it is better to search for a match by name from the beginning of a word, not only search for a combination of letters. But in some cases, your implementation could live. Change this please to search from the beginning

**Review by marynakorol — CHANGES_REQUESTED** [review]
take another quick eye at unresolved conversations and change points for pagination and filtering

File: js_task2/src/js/ModalWindow.js (hunk 17 above)
Comment (aotarasenko) [reply js_task2/src/js/ModalWindow.js]: Done

File: js_task2/index.html (hunk 16 above)
Comment (aotarasenko) [reply js_task2/index.html:77]: Done

File: js_task2/src/js/helpers.js (hunk 19 above)
Comment (aotarasenko) [reply js_task2/src/js/helpers.js:88]: Done

File: js_task2/src/scss/_variables.scss (hunk 7 above)
Comment (aotarasenko) [reply js_task2/src/scss/_variables.scss]: changed to $color-white

File: js_task2/index.html (hunk 1 above)
Comment (aotarasenko) [reply js_task2/index.html]: Done

File: js_task2/src/js/helpers.js (hunk 18 above)
Comment (aotarasenko) [reply js_task2/src/js/helpers.js:36]: Done

Comment (marynakorol) [reply js_task2/src/js/helpers.js:36]: image
still reproduced. There is no active first page when loaded

Comment (marynakorol) [reply js_task2/src/js/helpers.js:36]: image
left and right nav arrows should not take selected-page class, they can't be active themselves, only navigate to a page.

Comment (marynakorol) [reply js_task2/src/js/helpers.js:36]: and now the first page is empty 
image

File: js_task2/src/js/helpers.js (hunk 19 above)
Comment (marynakorol) [reply js_task2/src/js/helpers.js:88]: still reproduced

**Review by marynakorol — CHANGES_REQUESTED** [review]
take a look more attentionally  to the functionality or textto me if there are any additional points that need to be done to see the correct result

File: js_task2/src/js/helpers.js (hunk 18 above)
Comment (marynakorol) [reply js_task2/src/js/helpers.js:36]: something went wrong from my side, approving this PR according to video confirmation that functionality works correctly

**Review by marynakorol — APPROVED** [review]

**Pull request merged**